weto/src/main/java/fi/uta/cs/weto/actions/forum/ForumActions.java

```diff
@@ -422,6 +543,26 @@ public String action() throws Exception
                 .getValue(), tag.getId(), null, userIP).insert(conn);
       }
       addActionMessage(getText("forum.message.messageAdded"));
+      // Check if user has subscribed topic before.
+      ArrayList<Tag> subscriptionTags = Tag.selectByTaggedIdAndAuthorIdAndType(conn, taskId, userId, TagType.FORUM_TOPIC_SUBSCRIPTION.getValue());
+      Tag subscription = null;
+      for (int i = 0; i < subscriptionTags.size(); i++) {
+        if (subscriptionTags.get(i).getStatus().equals(topicId)) {
+          subscription = subscriptionTags.get(i);
+          break;
+        }
+      }
```

Comment:
Simplify with for-each
```suggestion
      for (Tag subscriptionTag : subscriptionTags) {
        if (subscriptionTag.getStatus().equals(topicId)) {
          subscription = subscriptionTag;
          break;
        }
      }
```